buildkite/scripts/archive/upgrade-script-check.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash
+
+# Archive Schema Upgrade Check Script
+#
+# This script verifies that when archive schema files are modified, the corresponding
+# upgrade script exists. This ensures database schema changes are properly handled
+# in production deployments.
+#
+# USAGE:
+#   upgrade-script-check.sh [OPTIONS]
+#
+# OPTIONS:
+#   -m, --mode MODE     Execution mode: 'conditional' or 'assert' (default: conditional)
```

Comment:
It seems to me `assert` is more like `verbose`, I suggest rename: ` conditional -> default` and `assert -> verbose`